resources/aws_credentials.rb

```diff
@@ -15,52 +15,53 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 #
+class CloudcliAwsCredentials < ::Chef::Resource
+  property :path, kind_of: String, name_property: true
+  property :profile, kind_of: String, default: 'default'
+  property :credential_params, kind_of: Hash, default: {}
+  property :owner, kind_of: String, default: 'root'
+  property :group, kind_of: String, default: 'root'
+  property :mode, kind_of: [String, Integer], default: 0600
 
-property :path, kind_of: String, name_property: true
-property :profile, kind_of: String, default: 'default'
-property :params, kind_of: Hash, default: {}
-property :owner, kind_of: String, default: 'root'
-property :group, kind_of: String, default: 'root'
-property :mode, kind_of: [String, Integer], default: 0600
+  action :create do
+    add_config_to_state(path, profile, credential_params)
 
-action :create do
-  add_config_to_state(path, profile, params)
+    template path do
+      owner new_resource.owner
+      group new_resource.group
+      mode new_resource.mode
+      source 'credentials.erb'
+      cookbook 'cloudcli'
+      variables(
+        :state_key => path
+      )
+      action :nothing
+    end
 
-  template path do
-    owner new_resource.owner
-    group new_resource.group
-    mode new_resource.mode
-    source 'credentials.erb'
-    cookbook 'cloudcli'
-    variables(
-      :state_key => path
-    )
-    action :nothing
+    # This allows end-users to use the resource multiple times
+    # but the template will only be run a single time.
+    #
+    # TODO: Refactor if this issue is resolved with a solution
+    #       https://github.com/chef/chef/issues/3426
+    log "create_template" do
```

Comment:
This is a great workaround for this, @alkalin3.  I will make sure to put this, hopefully eventual, refactoring in the backlog so that it doesn't get lost in the mix.